AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/QtPyScriptCanvasVariableManager.py

```diff
@@ -40,39 +41,118 @@ def __init__(self, sc_editor):
         variable_entry = pyside_utils.find_child_by_pattern(table_view, new_variable_type)
 
-        # Click on it to create variable
+        # Click on it to create variable and wait for the UI to respond
         pyside_utils.item_view_index_mouse_click(table_view, variable_entry)
+        actual_number_of_variables = graph_vars_table.model().rowCount(QtCore.QModelIndex())
+        variable_added = helper.wait_for_condition(lambda: (actual_number_of_variables == expected_number_of_variables),
+                                  WAIT_TIME_SEC_3)
+
+        assert variable_added, "New variable not added to graph variables list."
+
+    def __set_new_variable_name(self, new_variable_name: str) -> QtWidgets:
+        """
+        helper function for adding a new variable. sets the new variable's name. waits for the UI to respond afterwards
+
+        returns graph vars page object for future usage
+        """
+        # Get the widget with the newly added graph variable (currently selected)
+        graph_vars_page = self.variable_manager.findChild(QtWidgets.QWidget, GRAPH_VARIABLES_PAGE_QT)
+
+        # Set the new variable's name. Wait 3 seconds for the UI to respond
+        line_edit = graph_vars_page.findChild(QtWidgets.QLineEdit, "")
+        line_edit.insert(new_variable_name)
+        helper.wait_for_condition(lambda: line_edit.text() == new_variable_name, WAIT_TIME_SEC_3)
+
+        return graph_vars_page
+
+    def __get_variable_selection_index(self, graph_vars_page: QtWidgets.QWidget):
+        """
+        helper function for adding a new variable. gets the index object from the created graph variables widget then
+        emulates clicking somewhere else on the UI to remove focus from the graph variables . waits for the UI to
+        respond afterwards.
+
+        returns selection Qobject
+        """
+        selection_model = graph_vars_page.findChild(QtCore.QItemSelectionModel)
+        selection = selection_model.selection()
+        selection_model.clear()
+        helper.wait_for_condition(lambda: selection_model.hasSelection() is False, WAIT_TIME_SEC_3)
+
+        return selection
+
+    def select_variable_from_table(self, item_selection: QtCore.QItemSelection) -> None:
+        """
+        Function for selecting a variable from the table of created graph variables
+
+        params item_selection: the QtCore object reference to the variable in the table
+
+        returns None
+
+        """
+        graph_vars_page = self.variable_manager.findChild(QtWidgets.QWidget, GRAPH_VARIABLES_PAGE_QT)
+        selection_model = graph_vars_page.findChild(QtCore.QItemSelectionModel)
+        selection_model.select(item_selection, QtCore.QItemSelectionModel.Select)
+        helper.wait_for_condition(lambda: selection_model.hasSelection(), WAIT_TIME_SEC_3)
+
+        assert selection_model.selection() == item_selection, "Unable to select variable from variable manager table"
 
     def get_basic_variable_types(self):
```

Comment:
add return type

Reply:
added return type

Reply:
This one appears to have slipped through the update cracks.

Reply:
yep. I think the green highlighting threw me off and I missed this. fixed